### DISCUSSION TITLE
Can I use this on my code?

### DISCUSSION THREAD

**Discussion by massblabla** [armin-reichert/mazes#6] [General]
*see title*

**Comment by armin-reichert** [comment]
Hello,

it would be nice to give me some info on your project and yourself. 

As the license info for my project says, you are free to use it for whatever you want, as long as you adhere to the very permissive MIT license.

Regards
Armin Reichert

**Comment by massblabla** [comment]
Thank you for answering and sorry if I am too long,

I am just making a game where it has mazes in it, and I couldn't figure out how to make one, so I found this thingy.

about myslef i dont know how to describe it

**Comment by armin-reichert** [comment]
I would recommend using the code inside the mazes-simplified project. It has no dependencies to other projects (graph etc.) and is self-contained. Do you want to create mazes inside Minecraft?

**Comment by massblabla** [comment]
Actually, no, I am making a game. But thanks for the recommendation!

**Comment by massblabla** [comment]
> I would recommend using the code inside the mazes-simplified project. It has no dependencies to other projects (graph etc.) and is self-contained. Do you want to create mazes inside Minecraft?

Actually, let's say yes to the last question. It's similar to that

**Comment by armin-reichert** [comment]
So good luck. If you have any questions regarding the maze generation part, you are welcome.

**Comment by massblabla** [comment]
I have one question, how to make the walls of the maze in the cell, like
![image](https://github.com/user-attachments/assets/67923edf-9163-4cf0-b078-34f89ad75db3)

**Comment by armin-reichert** [comment]
This is the graph structure (blue circles = nodes, black lines = edges):

![maze-graph-image](https://github.com/user-attachments/assets/f472c306-f0bd-4ff7-8542-cba7965c21be)

**Comment by massblabla** [comment]
Sorry, I don't understand.

**Comment by armin-reichert** [comment]
What exactly? Aren't you familiar with graph-theoretic notions (graph, node, edge, tree, spanning tree) or something else?

The blue circles, the "nodes" of the graph (only some of them are drawn, the rest of the graph is similar) are arranged on a grid (rows, columns). Initially there are no connections (edges) between any circles. 

The maze generation algorithm computes connections (edges) between these circles (only between neighbors and only towards north, east, south or west neighbor). When the maze generator has finished, each circle can be reached from each other circle via a **unique** path. That means in graph-theoretic notion, we have created a "spanning tree" for the grid.

And this spanning tree can then be used to draw such a picture as above or to create a 3D maze.
